xarray/plot/animate.py

```diff
@@ -0,0 +1,166 @@
+        If provided, create a new figure for the plot with the given size.
+        Height (in inches) of each plot. See also: ``aspect``.
+    ax : matplotlib axes object, optional
+        Axis on which to plot this figure. By default, use the current axis.
+        Mutually exclusive with ``size`` and ``figsize``.
+    x, y : string, optional
+        Dimensions or coordinates for x, y axis.
+        Only one of these may be specified.
+        The other coordinate plots values from the DataArray on which this
+        plot method is called.
+    xscale, yscale : 'linear', 'symlog', 'log', 'logit', optional
+        Specifies scaling for the x- and y-axes respectively
+    xticks, yticks : Specify tick locations for x- and y-axes
+    xlim, ylim : optional
+        Specify x- and y-axes limits.
+    xincrease : None, True, or False, optional
+        Should the values on the x axes be increasing from left to right?
+        if None, use the default for the matplotlib function.
+    yincrease : None, True, or False, optional
+        Should the values on the y axes be increasing from top to bottom?
+        if None, use the default for the matplotlib function.
+    **kwargs : optional
+        Additional arguments to animatplot.blocks.Line
+
+    """
+
+    from animatplot.blocks import Line, Title
+    from animatplot.animation import Animation
+
+    row = kwargs.pop('row', None)
+    col = kwargs.pop('col', None)
+    if row or col:
+        raise NotImplementedError("Animated FacetGrids not yet implemented")
+
+    hue = kwargs.pop('hue', None)
+    if hue:
+        raise NotImplementedError("Animating multiple lines at once is not yet"
+                                  "implemented")
+
+    _check_animate(darray, animate)
+    darray = _transpose_before_animation(darray, animate)
+
+    ndims = len(darray[animate].dims)
+    if ndims > 2:
+        raise ValueError('Animated line plots are for 2- or 3-dimensional '
+                         'DataArrays. Passed DataArray has {ndims} '
+                         'dimensions'.format(ndims=ndims+1))
+
+    # Ensures consistency with .plot method
+    figsize = kwargs.pop('figsize', None)
+    aspect = kwargs.pop('aspect', None)
+    size = kwargs.pop('size', None)
+    ax = kwargs.pop('ax', None)
+    hue = kwargs.pop('hue', None)
+    x = kwargs.pop('x', None)
+    y = kwargs.pop('y', None)
+    linestyle = kwargs.get('linestyle', '')
+    xincrease = kwargs.pop('xincrease', None)  # default needs to be None
+    yincrease = kwargs.pop('yincrease', None)
+    xscale = kwargs.pop('xscale', None)  # default needs to be None
+    yscale = kwargs.pop('yscale', None)
+    xticks = kwargs.pop('xticks', None)
+    yticks = kwargs.pop('yticks', None)
+    xlim = kwargs.pop('xlim', None)
+    ylim = kwargs.pop('ylim', None)
+    _labels = kwargs.pop('_labels', True)
+
+    ax = get_axis(figsize, size, aspect, ax)
+    xplt_val, yplt_val, hueplt, xlabel, ylabel, huelabel = \
```

Comment:
Does `animate` work with `pd.Interval` step plots?

Reply:
I have no idea. In theory then as long as `xplt_val` and `yplt_val` get set correctly then it should? 

Have you got an example, issue, or point in the code somewhere I can look to see what a `pd.Interval` step plot consists of?

Reply:
This is the bit that tests it:
https://github.com/pydata/xarray/blob/a84ea12d34a4d1a542a34abbcf2feb3937a6a282/xarray/tests/test_plot.py#L460

Reply:
If it doesn't work; just raise an informative error...